### DISCUSSION TITLE
AttributeError: type object 'EnglishDefaults' has no attribute 'create_tokenizer'

### DISCUSSION THREAD

**Discussion by bycloudai** [explosion/spaCy#7398] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour
```
>>> from spacy.lang.en import English
>>> tokenizer = English().Defaults.create_tokenizer()
```
and I get this error

![image](https://user-images.githubusercontent.com/29135514/110334683-fd9a6500-805d-11eb-95c2-9be0c24618e6.png)

How can I make this work without running into an error?
My system default is English, but I also use Chinese.

I'll put this here if it helps. 
```
PS C:\WINDOWS\system32> dism /online /get-intl
...
Default system UI language : en-US
System locale : en-US
...
Installed language(s): en-US
  Type : Fully localized language.
Installed language(s): zh-TW
  Type : Partially localized language, MUI type.
  Fallback Languages en-US
```
![image](https://user-images.githubusercontent.com/29135514/110639642-eb016680-81ea-11eb-95c7-62048fb1fe43.png)

tag: @billagee

## Your Environment

* Operating System: Windows 10 insider build 21327
* Python Version Used: 3.7
* spaCy Version Used: 3.0.3
* Environment Information: WSL2 Ubuntu 18.04

**Comment by polm** [comment, marked as answer]
So that code works with spacy 2.3.0, but not with v3. The `create_tokenizer` function is basically an internal implementation detail and not really a public API. 

What are you trying to do exactly? If you just want to use spaCy as a tokenizer, normally you could use `spacy.blank("en")` or use a small model and disable pipelines. 

Also, as a note, spaCy doesn't use your locale settings for anything, so the language your computer is in shouldn't affect how spaCy behaves.

**Reply by ines** [reply]
To add to this for completeness, the v3-compatible translation using the public API would probably be this (which already worked in v2 as well):

```python
nlp = English()
tokenizer = nlp.tokenizer
```

(And I guess in theory, you could end up with weird unicode issues caused by the locale settings, e.g. #1684 – although I'm pretty sure that was related to the `regex` library, which spaCy isn't using anymore. And I haven't seen any problems like it come up in a long time.)

**Comment by polm** [comment]
Since this is a usage-related question, I'm going to move it to a discussion - you'll see that this issue is closed, but if you click on the link at the bottom you can follow up in the discussions forum.

**Comment by TobiasUhmann** [comment]
Just as a note: I also ran into the problem, because the official  docs (https://spacy.io/api/tokenizer) contain it as an example:

```python
# Construction 1
from spacy.tokenizer import Tokenizer
from spacy.lang.en import English
nlp = English()
# Create a blank Tokenizer with just the English vocab
tokenizer = Tokenizer(nlp.vocab)

# Construction 2
from spacy.lang.en import English
nlp = English()
# Create a Tokenizer with the default settings for English
# including punctuation rules and exceptions
tokenizer = nlp.Defaults.create_tokenizer(nlp)
```

Construction 1 returns a tokenizer that splits "Hello, world!" into "Hello," and "world!". Construction 2 seemed like an easy way to get a tokenizer that supports punctuation. But the `English().tokenizer` recommended above is even better.

**Reply by adrianeboyd** [reply]
Hmm, can you try a hard refresh of that webpage? On our end it looks like this was updated for v3 a while ago, but sometimes you do need to refresh or clear your cache to get the most recent version.

If you go to https://v2.spacy.io/api/tokenizer, you can still see the v2 docs with the older version of this example.

**Reply by TobiasUhmann** [reply]
Yeah, that worked. Did an "Empty Cache and Hard Reload" with the Chrome Dev Tools open and now the docs tell me to use `tokenizer = nlp.tokenizer`. 👍
